### ISSUE TITLE
Support stacks from different regions within the same app

### ISSUE THREAD

**Issue by fwang** [serverless-stack/serverless-stack#1299]
---
Requests:
- https://serverless-stack.slack.com/archives/C01JG3B20RY/p1643119080123400

**Comment by fwang** [comment]
Hey @jpmtrabbold, it's not on our immediate roadmap.

Imagine you need to deploy 3 stacks, `A`, `B`, and `C`. `A` needs to be deployed to `us-east-1`, `B` and `C` need to be deployed to `eu-west-1`. Here's the current recommended way:

```ts
if (app.region === "us-east-1") {
  app.stack(StackA);
}
else if (app.region === "eu-west-1") {
  app.stack(StackB).stack(StackC);
}
```

And to deploy, run `sst deploy --region=us-east-1 && sst deploy --region=eu-west-1`

By keeping all stacks in the "same app", the app name is the same for all stacks. When SST adds support for multi-region deployments, you can seamlessly remove the if conditions.

**Comment by jayair** [comment]
Yeah we'll need to figure out a better way to do this.

**Comment by garretcharp** [comment]
Just wanted to add what my ideal setup would be for multi-region:

Database Stack (deploys from a single region only):
- Global Dynamo DB Table in all regions I want to deploy
- Algolia Indexes (uses custom resources)

API Stack (Deploys in X regions):
- Queues
- Appsync API (connected to DynamoDB/Queues for this region)

Event Stack (Deploys in X regions):
- Eventbridge
- Create Dynamo triggers for the global table in this region and pushed to Eventbridge in this region
- Eventbridge trigger to push dynamo update to the appsync api in this region
- Any other triggers for backend events that happen from Dynamo events

Route53 stack 1:
- Creates a latency-based record to route to all the appsync apis
- Creates health checks to stop routing a specific region if it dies

SSR Frontend Stack (Deploys in X regions :: Not Lambda@Edge):
- Remix site 1
- Remix site 2
Will use Dynamo in current region for SSR data, and calls the global appsync url for client requests/subscriptions

Static Frontend Stack (Deploys in 1 region? I assume these are just global by default):
- React site 1: uses the global appsync api url for requests/subscriptions

Route53 stack 2:
- Creates a latency-based record to route to all SSR frontends
- Creates the health checks
- Creates the static frontend records

If I could set this up by just doing something like this it would be really cool:
```ts
app.stack(DatabaseStack)

for (const region of regions) {
    app.stack(APIStack, { region })
    app.stack(EventStack, { region })
}

app.stack(Route53APIRecordsStack)
app.stack(StaticFrontend)

for (const region of regions) {
    app.stack(SSRFrontendStack, { region })
}

app.stack(Route53FrontendRecordsStack)
```


In addition, would be really nice if I can reference Route53 across accounts. My work setup deploys every stage into different AWS accounts but all share the same domain. Currently, we manually create hosted zones in each account and add needed NS records on the account the domain lives in.


I'm not an expert on multi-region deployments as I have never done one so I am sure I am probably missing things here and this is very heavily simplified 😅 but if it was this easy to do multi-region active-active setups it would be super nice for when AWS has single-region failures and to generally make things faster for everyone globally.

And I did intentionally not use Lambda@Edge for the SSR sites mostly given all of our requests will hit the database multiple times so Lambda@Edge doesn't speed things up and makes it a tiny bit more complicated to try and figure out which database would be closest to the user and would probably need to add retry logic to use other regions to load data when using L@E instead of route53 handling that with the health check.

**Comment by TheUncharted** [comment]
If I remember what I did for global table for dynamodb a syntax like this would avoid the for loop

regions: [ "us-east-1", "us-east-2", "us-west-2" ] 

as attribute of your construct maybe ?

**Comment by garretcharp** [comment]
> If I remember what I did for global table for dynamodb a syntax like this would avoid the for loop
> 
> regions: [ "us-east-1", "us-east-2", "us-west-2" ]
> 
> as attribute of your construct maybe ?

Yes setting up a global table is easy (and not even in the loop). This part can indeed already be done with SST. But having a global table is pointless for me until SST supports multi-region for other aspects. I think it could be possible to fully setup what I would want but it would be very convoluted to setup properly.

**Comment by khuezy** [comment]
I'd like to deploy `Service` to `us-east-1` and `eu-west-1`, using the recommended way: `sst deploy --region=us-east-1 && sst deploy --region=eu-west-1`, how would I reference the CF distribution domain URL for a third stack that sets up the global `Route53 Cname`?

I suppose I can add another step to read from `.sst/outputs.json` but the 2nd deployment would override the first's output.

@fwang what's the status on multi-region support?

**Comment by jayair** [comment]
It's pretty manual at the moment I think for us to do multi-region properly, we'll need to step outside CDK.

**Comment by patsuitestudios** [comment]
Sounds like SST v3 has effectively "stepped outside CDK", does this mean there's improved support for multi-region deploys?

**Comment by jayair** [comment]
Yeah in v3 there is https://sst.dev/docs/providers#multi-region
